editor/src/clj/editor/atlas.clj

```diff
@@ -758,12 +758,22 @@
    page     :- types/Int32
    geometry :- s/Any])
 
+(defn rotate-vertices-90-cw [vertices]
+  (mapcat (fn [[x y]]
+            [y (- x)])
+          (partition 2 vertices)))
+
 (g/defnk produce-image-path->rect
   [layout-size layout-rects texture-set]
   (let [[w h] layout-size
         geometries (:geometries texture-set)]
     (into {} (map (fn [{:keys [path x y width height index page]}]
-                    [path (->AtlasRect path x (- h height y) width height page (get geometries index))]))
+                    (let [geometry (get geometries index)
+                          rotated-vertices (if (:rotated geometry)
+                                             (rotate-vertices-90-cw (:vertices geometry))
+                                             (:vertices geometry))]
+                      [path (->AtlasRect path x (- h height y) width height page
+                                         (assoc geometry :vertices rotated-vertices))])))
```

Comment:
Hmm, is this the right place for the change? I mean, is this the only place where the vertices from `texture-set` need to be rotated? Could it be that we want the vertices to be rotated everywhere once we have them? In that case, we might want to do it in `editor.pipeline.texture-set-gen/atlas->texture-set-data`, for example...

Reply:
It's a good question. I didn't find a better place  + here it's cached. But I agree, it might be not the best place and I just still have "fear" to introduce new entities in the editor code.

Reply:
Back when I was fixing various issues in this area, I tried to unify the various code paths that dealt with vertex geometry. To make sure it is picked up everywhere, I think we may want to make this change in the `texture-set/make-anim-data` code path, specifically in `->anim-frame-from-geometry` called from the `->anim-data` function. Here we deal with the flip flags, so it might make sense to also deal with rotation here?

Unless of course we need to apply this rotation to "undo" an existing rotation for this specific use case? It's a bit hard to tell, because I don't know exactly what preparations are done to the geometry by the `TextureSetGenerator` class we share with Bob. 🤔

Reply:
After investigation, I still think, that I use right place for it. 

The place @matgis recommended is the place where atlas produces its outputs, which used for all the components and the other places the atlas linked. 

But the only issue we have is representation in the atlas viewer itself. 
That's why I modified `image-path->rect` because the only place where it is used is `produce-image-scene` in `atlas`, So I fix the only place we have this issue, and it doesn't affect anything else.